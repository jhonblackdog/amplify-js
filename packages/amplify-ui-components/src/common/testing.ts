export const pixelThreshold = 10;
