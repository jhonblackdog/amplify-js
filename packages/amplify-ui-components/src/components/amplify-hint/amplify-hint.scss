.hint {
  color: var(--amplify-hint-color);
  font-family: var(--amplify-font-family);
  font-weight: 300;
  font-size: var(--amplify-third-font-size);
  margin-bottom: 2.625rem;
}
