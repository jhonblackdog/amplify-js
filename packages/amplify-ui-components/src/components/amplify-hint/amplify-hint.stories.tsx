import { storiesOf } from '@storybook/html';
import { text as textKnob } from '@storybook/addon-knobs';

storiesOf('amplify-hint', module).add('with text', () => {
  const text = textKnob('Text', 'Hint placeholder');
  return `<amplify-hint>${text}</amplify-hint>`;
});
