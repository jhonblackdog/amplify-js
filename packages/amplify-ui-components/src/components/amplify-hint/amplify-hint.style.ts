import { css } from 'emotion';

export const hint = css`
  color: var(--hint-color);
  font-family: var(--font-family);
  font-weight: 300;
  font-size: var(--third-font-size);
  margin-bottom: 2.625rem;
`;
