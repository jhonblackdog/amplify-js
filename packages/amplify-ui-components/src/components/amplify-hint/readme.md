# amplify-hint



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type      | Default |
| ---------- | ---------- | ----------- | --------- | ------- |
| `override` | `override` |             | `boolean` | `false` |


## Dependencies

### Used by

 - [amplify-form-field](../amplify-form-field)

### Graph
```mermaid
graph TD;
  amplify-form-field --> amplify-hint
  style amplify-hint fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
