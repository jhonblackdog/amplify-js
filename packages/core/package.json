{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
