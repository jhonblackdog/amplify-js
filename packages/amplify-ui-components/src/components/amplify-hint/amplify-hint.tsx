import { Component, h, Prop } from '@stencil/core';
import { hint } from './amplify-hint.style';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_HINT } from '../../common/constants';
@Component({
  tag: 'amplify-hint',
  shadow: false,
})

export class AmplifyHint {
  @Prop() override: boolean = false;

  render() {
    return (
      <div class={styleNuker(this.override, AMPLIFY_UI_HINT, hint)}>
        <slot />
      </div>
    );
  }
}
