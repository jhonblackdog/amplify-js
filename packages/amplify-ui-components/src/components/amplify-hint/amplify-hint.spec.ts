import { newSpecPage } from '@stencil/core/testing';
import { AmplifyHint } from './amplify-hint';

describe('amplify-hint spec:', () => {
  describe('Render logic ->', () => {
    it('renders with no hint text', async () => {
      const page = await newSpecPage({
        components: [AmplifyHint],
        html: `<amplify-hint></amplify-hint>`,
      });

      expect(page.root).toMatchSnapshot();
    });
    it('renders with hint text FOO', async () => {
      const page = await newSpecPage({
        components: [AmplifyHint],
        html: `<amplify-hint>FOO</amplify-hint>`,
      });

      expect(page.root).toMatchSnapshot();
    });
    it('renders without Emotion CSS class when overrideStyle is true', async () => {
      const page = await newSpecPage({
        components: [AmplifyHint],
        html: `<amplify-hint override-style='true'>FOO</amplify-hint>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
  describe('Component logic ->', () => {
    it('should have overrideStyle false by default', async () => {
      const hint = new AmplifyHint();
      expect(hint.overrideStyle).toBe(false);
    })
  });
});
