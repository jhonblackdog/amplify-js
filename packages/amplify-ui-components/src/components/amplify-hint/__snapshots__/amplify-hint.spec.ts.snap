// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-hint spec: Render logic -> renders with hint text FOO 1`] = `
<amplify-hint>
  <!---->
  <div class="amplify-ui--hint css-k9ji7b">
    FOO
  </div>
</amplify-hint>
`;

exports[`amplify-hint spec: Render logic -> renders with no hint text 1`] = `
<amplify-hint>
  <!---->
  <div class="amplify-ui--hint css-k9ji7b"></div>
</amplify-hint>
`;

exports[`amplify-hint spec: Render logic -> renders without Emotion CSS class when overrideStyle is true 1`] = `
<amplify-hint override-style="true">
  <!---->
  <div class="amplify-ui--hint">
    FOO
  </div>
</amplify-hint>
`;
